service/distros.go

```diff
@@ -126,7 +126,8 @@ func (uis *UIServer) modifyDistro(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	newDistro := oldDistro
+	newDistro := &distro.Distro{}
+	*newDistro = *oldDistro
```

Comment:
Does newDistro need to be a pointer at all? We may just be able to set it to *oldDistro directly (I'd also suggest adding a lil comment here so it's clear that this was intentional and not wonky)